### ISSUE TITLE
Authentication error

### ISSUE THREAD

**Issue by eberghoefer** [ExchangeCalendar/exchangecalendar#49]
Similar to what was mentioned in the #604 of the previous repository, with Exchange2010 server I still have an issue with the current 4.0.0-beta4 version. This fixes the problem of not showing up the password prompt, but still for me it doesn't work. If I enter the password I got a new window which then asks for a password, while trying to use my **previously entered password in plain text**! Of curse since such a user does not exist, it gets stuck, and finally aborts (after timeout?) with an authentication failed message. While this prevents me from getting updates on my calendars it is also a security issue, because of the plain text password will go into logs probably. 
Finally I fixed it by downgrading Thunderbird to 38 installing the old 3.9.0 version of the EWS provider and saved my new password and re-updated Thunderbird while keeping the old version of EWS provider. Seem to bu unusable in its current version.

Thunderbird: 52.2.1 (64-Bit)
Lightning: 5.4
EWS-provider: 4.0.0-beta4 (failed to login), while after using the described method to update my calendars and save the correct password in the pw manager again, it works now with 3.9.0.

**Comment by lunderhage** [comment]
I agree. For me, this addon just works the first time when installed. Then the above happens after restart of Thunderbird. Unusable at the moment.

**Comment by ansgar-radermacher** [comment]
I confirm the first observation with the 4.0.0-beta4 version. My 1st password entering is taken as a username (which is quite problematic). If I type my username during the first password request and then the right password, I get an error message (wrong pw or similar)

**Comment by maledo** [comment]
Unfortunately, I also can confirm the problem with beta4. But sometimes I got just the Client error even if the password wasn't set as user.

After updating to 4.0.0-beta5, I have another error: Failed to read `'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties.` Anyone else with the same behavior and idea to fix this? Unfortunately, the steps with downgrading to TB 38 etc. did not work for me either.

Thunderbird: 52.6.0
Lightning: 5.4
EWS: 4.0.0-beta4/4.0.0-beta5

**Comment by Trim** [comment]
Hello, we have moved the locales to a new namespace: I have to check, but I suspect the correct namespace should be echangecommon instead of exchangecalendar. Furthermore, you should try the build given in #148 as there's a bug with HTTP Basic authentication with the current 4.0.0-beta5

**Comment by maledo** [comment]
Thanks for the quick reply!

Yes, the new namespace seems to be common. But how could I change that? I tried with the new build (selfbuilt and yours), but still have the problem due to the namespace. I also reinstalled the plugin...

**Comment by Trim** [comment]
For the namespace, I have to update the code, but I don't have my PC right now. I'll update it this evening.

**Comment by Trim** [comment]
Oh, as you hadn't the issue with HTTP Basic Authentication, you can try current ec-4.0 branch: the PR #153 is already merged there and it fixes this issue with the namespace.

**Comment by maledo** [comment]
Thanks for the quick reply. I built the ec-4.0 branch. But now I get HTTP client error. Is there a branch with the namespace and the authentication fix?

**Comment by das-g** [comment]
> After updating to 4.0.0-beta5, I have another error: `Failed to read 'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties.` Anyone else with the same behavior and idea to fix this?

I'm getting that, too, when clicking the "Check server and mailbox" button in the "Create New Calendar" dialog.

Thunderbird: "Earlybird" 52.6.0 (64-bit)
Lightning 5.4.6
4.0.0-beta5 (also with the [build](https://cloud.adorsaz.ch/index.php/s/eTjpMdXnv8U8Mvh/download) from https://github.com/ExchangeCalendar/exchangecalendar/issues/148#issuecomment-362913073)

I'm following the instructions at https://www.openhsr.ch/app/thunderbird/#kalender (replacing the sample email addresses / username with my corresponding own ones at HSR.)

**Comment by das-g** [comment]
From the log (with credentials replaced with the example values from the openhsr.ch instructions)
```
POST 
XHR 
https://mmuster%40hsr%2Ech:GeHeim007@webmail.hsr.ch/ews/exchange.asmx [17ms]
POST 
XHR 
https://mmuster%40hsr%2Ech:GeHeim007@webmail.hsr.ch/ews/exchange.asmx [1ms]
POST 
XHR 
https://mmuster%40hsr%2Ech:GeHeim007@webmail.hsr.ch/ews/exchange.asmx [0ms]
Failed to read 'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties. Error: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIStringBundle.formatStringFromName]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: file:///home/das-g/.thunderbird/gx9bkvc7.default/extensions/exchangecalendar@extensions.1st-setup.nl/common/interface/global/mivFunctions.js :: _getString :: line 359"  data: no]
```

**Comment by arichiardi** [comment]
Can confirm this problem as well. Are there any known workarounds?

**Comment by finsterwalder** [comment]
I just fixed the problem for me.
I removed all the password stored for exchange users.
You can manage your password in the Thunderbird Preferences -> Security -> Saved Passwords
After doing so, I re-added my calendar and it works fine.

**Comment by SUSEsean** [comment]
Hey all, confirming the Failed to read 'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties. error. 

Using openSUSE Tumbleweed (4.16 kernel) with Thunderbird 52.7.0 (64-bit)

**Comment by ednarb29** [comment]
Confirming the error. 

> Failed to read 'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties

Ubuntu 18.04
Thunderbird 52.7.0 (64 bit)
Lightning 5.4
Exchange 2016

@finsterwalder The purposed fix did not work for me.

Any progress?

**Comment by ednarb29** [comment]
> Oh, as you don't have the issue with HTTP Basic Authentication, you can try current ec-4.0 branch: the PR #153 is already merged there and it fixes this issue with the namespace.

@Trim Unfortunately this fix did not work for me either.

Ubuntu 18.04
Thunderbird 52.7.0 (64 bit)
Lightning 5.4
Exchange 2016

**Comment by qsantos** [comment]
I just started encountering this problem today, and none of the fixes listed here work. I can no longer access my EWS calendar and instead get the following error message:

```
Failed to read 'ecErrorServerAndMailboxCheck' from chrome://exchangecalendar/locale/calExchangeCalendar.properties.
```

**Comment by eliSodeglo** [comment]
having the same problem with TB 52.9.1 (64bit) on Debian 9.5 Stretch and couldn't fix it with the proposed tip
(found this related issues: #165 #174 )
